### ISSUE TITLE
Tax rate for billing country of logged-in user ignored before checkout, should be detected and applied

### ISSUE THREAD

**Issue by glueckpress** [woothemes/woocommerce#7459]
### Settings

* Taxes:
   - **Prices entered with tax:** _Yes, I will enter prices inclusive of tax_
   - **Calculate tax based on:** _Customer billing address_
   - **Default customer address:** _Shop base address_
   - **Display prices [in shop/during cart/checkout]:** _Including tax_
* Country:
   - **Shop base location:** _Germany_ (Standard tax rate configured: 19%)
   - **Billing country of logged-in user:** _Austria_ (Standard tax rate configured: 20%)
* Product:
   - **Regular price:** _€119.00_

Tax rates configured:
![bildschirmfoto 2015-02-17 um 12 13 56](https://cloud.githubusercontent.com/assets/308422/6227781/4e958118-b6a4-11e4-8409-b3db15df6fc3.png)

Tax settings configured:
![bildschirmfoto 2015-02-17 um 12 02 35](https://cloud.githubusercontent.com/assets/308422/6227815/b1e3e480-b6a4-11e4-8812-6f87064be084.png)
![bildschirmfoto 2015-02-17 um 12 14 30](https://cloud.githubusercontent.com/assets/308422/6227823/bd11f61c-b6a4-11e4-8503-281131b32d82.png)

User billing address:
![bildschirmfoto 2015-02-17 um 12 05 45](https://cloud.githubusercontent.com/assets/308422/6227828/c6dd7446-b6a4-11e4-9897-559f17a24103.png)

------

### Expectation

**For a logged-in user whose billing country is `$country`, WooCommerce should attempt to apply tax rates configured for `$country` to all prices.**

------

#### Product loop and single product:

Price displayed as entered by shop admin (assumed to include 19% shop base tax rate in our specific scenario of settings).

![bildschirmfoto 2015-02-17 um 12 16 00](https://cloud.githubusercontent.com/assets/308422/6227731/cf39db6c-b6a3-11e4-8001-0d131fa30593.png)

#### Cart:

Tax shown as included is German 19% (shop base tax rate), despite the fact I’m logged-in as a user from Austria (20%):

![bildschirmfoto 2015-02-17 um 12 16 26](https://cloud.githubusercontent.com/assets/308422/6227733/d59c64b6-b6a3-11e4-8d96-6abc3016bc24.png)

#### Checkout

Finally, the tax rate configured for my (the user’s) billing country is applied, resulting in prices changing unnecessarily late for me:

![bildschirmfoto 2015-02-17 um 12 17 39](https://cloud.githubusercontent.com/assets/308422/6227795/8196e818-b6a4-11e4-82a0-5e1a25b946f1.png)

------

### TL;DR

According to settings described above, **at checkout** WC would calculate the net price for our product by removing a tax amount of (in this case) 19% shop base country tax:

```php
// $price_incl_tax = product price as configured, 
// supposed to include standard tax of shop base

$price_excl_tax = $price_incl_tax / ( 100 + $shop_base_standard_tax_rate ) * 100;

// €119.00 / ( 100 + 19 ) * 100 = €100.00
```

However, if a user is already logged-in, shouldn’t that calculation take place **earlier than checkout**?
When WC is able to retrieve a billing country from the user profile, what’s the point of showing that user product prices that include a non-applicable tax rate for her/his country, and then re-calculate taxes during checkout?
